qadence/analog/utils.py

```diff
@@ -130,3 +132,21 @@ def rot_generator(block: ConstantAnalogRotation) -> AbstractBlock:
     x_terms = (omega / 2) * add(cos(phase) * X(i) - sin(phase) * Y(i) for i in support)
     z_terms = delta * add(N(i) for i in support)
     return x_terms - z_terms  # type: ignore[no-any-return]
+
+
+def add_pattern(register: Register, pattern: Union[AddressingPattern, None]) -> AbstractBlock:
+    support = tuple(range(register.n_qubits))
+    if pattern is not None:
+        max_amp = pattern.max_amp
+        max_det = pattern.max_det
+        weights_amp = pattern.weights_amp
+        weights_det = pattern.weights_det
+    else:
+        max_amp = 0.0
+        max_det = 0.0
+        weights_amp = {i: 0.0 for i in support}
+        weights_det = {i: 0.0 for i in support}
+
+    p_drive_terms = (1 / 2) * max_amp * add(X(i) * weights_amp[i] for i in support)
+    p_detuning_terms = -max_det * add(0.5 * (I(i) - Z(i)) * weights_det[i] for i in support)
```

Comment:
Can you use `N(i)` instead? It should be equivalent and more clear.

Reply:
It seems that no - since `N` operation doesn't have a generator or generator eigenvalues defined. Because of this when calculating derivatives an exception is thrown.

Reply:
The real reason for using `I` and `Z` instead of `N` is that when we have a prefactor in front of `N` it is treated as a `ScaleBlock` and gets assigned a uuid. This in turn means that `uuid_to_eigen` function tries to get the eigenvalues of `N` that it doesn't have.

Reply:
Hmm strange, I thought I added the eigenvalues to N